packages/core-flows/src/product/steps/remove-variant-pricing-link.ts

```diff
@@ -0,0 +1,49 @@
+import { Modules } from "@medusajs/modules-sdk"
+import { ILinkModule } from "@medusajs/types"
+import { ContainerRegistrationKeys } from "@medusajs/utils"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+type StepInput = {
+  variant_ids: string[]
+}
+
+export const removeVariantPricingLinkStepId = "remove-variant-pricing-link"
+export const removeVariantPricingLinkStep = createStep(
+  removeVariantPricingLinkStepId,
+  async (data: StepInput, { container }) => {
+    const remoteLink = container.resolve(ContainerRegistrationKeys.REMOTE_LINK)
+    const linkModule: ILinkModule = remoteLink.getLinkModule(
+      Modules.PRODUCT,
+      "variant_id",
+      Modules.PRICING,
+      "price_set_id"
+    )
+
+    const links = (await linkModule.list(
+      {
+        variant_id: data.variant_ids,
+      },
+      { select: ["id", "variant_id", "price_set_id"] }
+    )) as { id: string; variant_id: string; price_set_id: string }[]
+
+    await remoteLink.delete(links.map((link) => link.id))
+    return new StepResponse(void 0, links)
+  },
+  async (prevData, { container }) => {
+    if (!prevData?.length) {
+      return
+    }
+
+    const remoteLink = container.resolve(ContainerRegistrationKeys.REMOTE_LINK)
+    await remoteLink.create(
```

Comment:
You might have to do a restore here in case any of the steps fail. You still want to have the prices reverted upon compensation.

Reply:
@riqwan this is the part I am not sure how exactly it works, let me leave it as-is for a second.